lib/auth/user_gateway.dart

```diff
@@ -12,8 +12,14 @@ class UserGateway {
   Future<void> requestEmailVerification() =>
       _post('sendOobCode', {'requestType': 'VERIFY_EMAIL'});
 
-  Future<User> getUser() async {
-    var map = await _post('lookup', {});
+  Future<User> getUser({String uid}) async {
+    var map = await _post(
+        'lookup',
+        uid == null
+            ? {}
+            : {
+                'localId': [uid]
+              });
```

Comment:
Why are you specifying the id?

Reply:
So we can do getUser(). We want to verify if the incoming user has a valid token, right? So in order to get the user object, we have to give it the uid we're receiving.

Reply:
Ok, I see, but wouldn't it be easier to make the request using the token you got from the user?

Reply:
In cases like a REST api, you feed the user token directly in the request.

Reply:
Yes, from the token store. What's your point?

Reply:
I don't understand what you mean. If some random person that just signed up to my app and registered in firebase auth, then that same person sends a request to my API with their token, I've never seen that token before, the token store would not have this in memory. Am I missing something?

Reply:
Their token is in their request. It's the token that you're trying to validate.

Reply:
verifyToken is NOT dependant on firebase auth initialization.

In the backend, firebase auth is initialized with the service account, you get admin firestore as a result and admin access to whatever the sdk provides.
We have use cases were you do NOT initialize firebase auth with the requester's token.

This is how it is done in the official java admin sdk as well. 
So it depends if you want people to initialize a firebase auth instance with a volatile token store whenever someone wants to do verifyToken.

This implementation is simpler in my opinion, but if you really want people to have to make firebase auth instances, I'll make it so... Although there currently isn't a signInWithToken method in the auth gateway

Reply:
This goes to the review you made below this one, where you want me to make a volatile tokenstore.

I was confused at first because that's not really required. getUser can be a static method that doesn't need an initialized firebase auth, right?

Reply:
Then create a constructor that takes a token as initialiser. Make a single call on behalf of the user with their token. If it passes the token is good, if not, it's now.
How much more simple can this get?

Reply:
I mean, this is just as simple... But okay if that's what you want.

Reply:
I'm going to keep this for now, we do not need to sign in the user. We just need an initialized firebase auth and look up any user no matter the id. This is simpler.